### ISSUE TITLE
FR: Make 'export notebook' functionality more discoverable

### ISSUE THREAD

**Issue by Gitoffthelawn** [orgzly/orgzly-android#789]
Exporting Orgzly notebooks is a key function for users who don't sync.  From what I can tell, it may be the only way for such users to backup Orgzly data without root.  (Is that correct?)

Finding the export function in the current UI is not trivial.  I recommend making it more discoverable.  One such method would be adding it to the Settings menu.

**Comment by nevenz** [comment]
> Exporting Orgzly notebooks is a key function for users who don't sync. From what I can tell, it may be the only way for such users to backup Orgzly data without root. (Is that correct?)

Yes, although setting a "Directory" repository type would be easier. If there's nothing writing to that directory on the device, using "Sync" is basically the same as exporting all notebooks. If exporting *all* notebooks is fine with you.

> Finding the export function in the current UI is not trivial. I recommend making it more discoverable. One such method would be adding it to the Settings menu.

It's not really a preference to go to Settings. I think long-clicking the notebook and then checking all options in the menu is pretty standard on Android.

Maybe it's something that could be documented though. Perhaps a new "Notebooks" section in [Getting Started with Orgzly](https://github.com/orgzly/documentation/blob/master/android/getting-started.org) notebook.

**Comment by Gitoffthelawn** [comment]
> Yes, although setting a "Directory" repository type would be easier. If there's nothing writing to that directory on the device, using "Sync" is basically the same as exporting all notebooks. If exporting all notebooks is fine with you.

Thanks.  Good info.  May be worth documenting.

> It's not really a preference to go to Settings. I think long-clicking the notebook and then checking all options in the menu is pretty standard on Android.

Hmmm... I'm going to somewhat disagree, but only because I find that I rarely need to long-press anything in Android, except when performing a multi-select operation.  I think that is why I perceive this as an issue: because even to export a single notebook, you have to long-press it.  The options are completely hidden from the user unless they long-press on a notebook, which isn't really an obvious thing to do.

Generally, it's best to avoid forcing the user to long-press on all UI elements in order to discover otherwise hidden functionality.  (It's one reason I think the UX of macOS is overrated, because so much important functionality is hidden from the user... but I digress...)

> Maybe it's something that could be documented though. Perhaps a new "Notebooks" section in Getting Started with Orgzly notebook.

Yes, I think that would be helpful.  Of course, most people probably don't read the docs, but that's a broader issue.  I like your idea of putting it in "Getting Started", because if someone reads just one section of the documentation, that's likely going to be it. :)

**Comment by nevenz** [comment]
> Hmmm... I'm going to somewhat disagree, but only because I find that I rarely need to long-press anything in Android, except when performing a multi-select operation. I think that is why I perceive this as an issue: because even to export a single notebook, you have to long-press it. The options are completely hidden from the user unless they long-press on a notebook, which isn't really an obvious thing to do.

Yeah, I agree about long-press being slow - I'm not a fan of it either. But export was never really meant to be used *regularly*, which is why I don't have an issue being where it is now I guess.

I'd highly suggest using Sync instead for backup.
